public/jira/derived/work-status/work-status.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * This module is repsonsible for determining the correct workType ("design", "dev", "qa", "uat")
+ * and statusType ("qa", "uat", "todo", "done", "blocked") for an issue.
+ */
+export const workType = ["design", "dev", "qa", "uat"] as const;
+export const workTypes = workType;
+
+export const workflowHappyPath = ["todo", "design", "dev", "qa", "uat", "done"];
+export const workflowUnhappyStatuses = ["blocked"];
+
+const inQAStatus = { qa: true, "in qa": true, "qa complete": true };
+const inPartnerReviewStatus = { "partner review": true, uat: true };
+const inDoneStatus = { done: true, cancelled: true };
+const blockedStatus = { blocked: true, delayed: true };
+
+export const inPartnerReviewStatuses = Object.keys(inPartnerReviewStatus);
+export const inIdeaStatus = { idea: true, "to do": true, open: true };
+export const inIdeaStatuses = Object.keys(inIdeaStatus);
+
+type WorkType = (typeof workType)[number];
+
+type Status =
+  | keyof typeof inQAStatus
+  | keyof typeof inPartnerReviewStatus
+  | keyof typeof inIdeaStatus
+  | keyof typeof inDoneStatus
+  | keyof typeof blockedStatus
+  | (string & {});
+
+type StatusCategory = "qa" | "uat" | "todo" | "done" | "blocked";
+
+export const statusCategoryMap = (function () {
+  const items = [
+    ["qa", inQAStatus],
+    ["uat", inPartnerReviewStatus],
+    ["todo", inIdeaStatus],
+    ["done", inDoneStatus],
+    ["blocked", blockedStatus],
+  ] as const;
+
+  const statusCategoryMap = {} as Record<Status, StatusCategory>;
+
+  for (let [category, statusMap] of items) {
+    for (let prop in statusMap) {
+      const status = prop;
+      statusCategoryMap[status] = category;
+    }
+  }
+
+  return statusCategoryMap;
+})();
+
+export function getStatusTypeDefault(issue: { status?: string }): StatusCategory | "dev" {
+  const statusCategory = statusCategoryMap[toLowerCase(issue?.status || "")];
+  if (statusCategory) {
+    return statusCategory;
+  } else {
+    return "dev";
+  }
+}
+
+function toLowerCase(str: string): string {
+  return str.toLowerCase();
+}
+
+const workPrefix = workType.map((wt) => wt + ":");
+
+function getWorkTypeDefault(normalizedIssue: { summary?: string; labels?: string[] }): WorkType {
+  let wp = workPrefix.find((wp) => (normalizedIssue?.summary || "").toLowerCase().indexOf(wp) === 0);
+
+  if (wp) {
+    return wp.slice(0, -1) as WorkType;
+  }
+
+  wp = workType.find((wt) => normalizedIssue.labels?.map(toLowerCase).includes(wt));
```

Comment:
nitpick: Also not related to the scope of this PR :D 
But maybe we can replace this logic to the use the string's `toLowerCase` instead of our own custom function above that basically just calls `toLowerCase` directly

```
  wp = workType.find((wt) =>
    normalizedIssue.labels?.some((label) => label.toLowerCase() === wt)
  );
  ```

Reply:
Ha this was another one of those, maybe I should maybe i shouldn't its a small change. Since you pointed it out too i think its worth updating. One sec